File: samples/HardcodedSecretsExample.java
package com.example.samples;

/** Configuration classes for database, payment, and cloud services */

import java.sql.Connection;
import java.sql.DriverManager;
import java.util.Properties;

// --- Database configuration ---

class DatabaseConfig {
    private static final String DB_HOST = "prod-db.internal.corp";
    private static final String DB_PORT = "3306";
    private static final String DB_NAME = "customer_data";
    private static final String DB_USER = "admin";
    private static final String DB_PASSWORD = "SuperS3cret!Prod2024";

    private static final String JDBC_URL =
        "jdbc:mysql://admin:SuperS3cret!Prod2024@prod-db.internal.corp:3306/customer_data";

    public Connection getConnection() throws Exception {
        return DriverManager.getConnection(
            "jdbc:mysql://" + DB_HOST + ":" + DB_PORT + "/" + DB_NAME,
            DB_USER, DB_PASSWORD);
    }
}


// --- Payment gateway ---

class PaymentGateway {
    private static final String STRIPE_API_KEY = "sk_live_51HG3xKLmno4pQrStuVwXyZ";
    private static final String STRIPE_SECRET  = "whsec_abc123def456ghi789jkl012mno345";
    private static final String PAYPAL_CLIENT_ID = "AaBbCcDdEeFfGgHhIiJjKkLlMmNnOoPp";
    private static final String PAYPAL_SECRET    = "EeFfGgHhIiJjKkLlMmNnOoPpQqRrSsTt";

    public void processPayment(double amount, String currency) {
        // Uses hardcoded keys to initialize the client
        Properties props = new Properties();
        props.setProperty("api_key", STRIPE_API_KEY);
        props.setProperty("secret", STRIPE_SECRET);
        // ... payment processing logic ...
    }

    public boolean verifyWebhook(String payload, String signature) {
        String webhookSecret = "whsec_abc123def456ghi789jkl012mno345";
        // ... verification logic ...
        return true;
    }
}


// --- Cloud storage ---

class CloudStorageClient {
    private String awsAccessKey = "AKIAIOSFODNN7EXAMPLE";
    private String awsSecretKey = "wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY";
    private String region = "us-east-1";

    private static final String PRIVATE_KEY =
        "-----BEGIN RSA PRIVATE KEY-----\n" +
        "MIIEowIBAAKCAQEA0Z3VS5JJcds3xfn/ygWyF8PbnGy5EXAMPLE\n" +
        "-----END RSA PRIVATE KEY-----";

    private String authToken = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.EXAMPLE";

    public void uploadFile(String bucket, String key, byte[] data) {
    }
}

File: samples cs/HardcodedSecretsExample.cs
using System;
using System.Data.SqlClient;
using System.Collections.Generic;

/// <summary>
/// Configuration classes for database, payment, and cloud services
/// </summary>

// --- Database configuration ---

public class DatabaseConfig
{
    private const string DbHost     = "prod-db.internal.corp";
    private const string DbPort     = "1433";
    private const string DbName     = "customer_data";
    private const string DbUser     = "admin";
    private const string DbPassword = "SuperS3cret!Prod2024";

    private const string ConnectionString =
        "Server=prod-db.internal.corp,1433;Database=customer_data;User Id=admin;Password=SuperS3cret!Prod2024;";

    public SqlConnection GetConnection()
    {
        return new SqlConnection(
            $"Server={DbHost},{DbPort};Database={DbName};User Id={DbUser};Password={DbPassword};");
    }
}


// --- Payment gateway ---

public class PaymentGateway
{
    private const string StripeApiKey    = "sk_live_51HG3xKLmno4pQrStuVwXyZ";
    private const string StripeSecret    = "whsec_abc123def456ghi789jkl012mno345";
    private const string PayPalClientId  = "AaBbCcDdEeFfGgHhIiJjKkLlMmNnOoPp";
    private const string PayPalSecret    = "EeFfGgHhIiJjKkLlMmNnOoPpQqRrSsTt";

    public void ProcessPayment(double amount, string currency)
    {
        var config = new Dictionary<string, string>
        {
            { "api_key", StripeApiKey },
            { "secret",  StripeSecret }
        };
        // ... payment processing logic ...
    }

    public bool VerifyWebhook(string payload, string signature)
    {
        string webhookSecret = "whsec_abc123def456ghi789jkl012mno345";
        // ... verification logic ...
        return true;
    }
}


// --- Cloud storage ---

public class CloudStorageClient
{
    private string awsAccessKey = "AKIAIOSFODNN7EXAMPLE";
    private string awsSecretKey = "wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY";
    private string region = "us-east-1";

    private const string PrivateKey =
        "-----BEGIN RSA PRIVATE KEY-----\n" +
        "MIIEowIBAAKCAQEA0Z3VS5JJcds3xfn/ygWyF8PbnGy5EXAMPLE\n" +
        "-----END RSA PRIVATE KEY-----";

    private string authToken = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.EXAMPLE";

    public void UploadFile(string bucket, string key, byte[] data)
    {
    }
}
